src/helpers/validateRegisterFields.js

```diff
@@ -0,0 +1,54 @@
+import { validateDate } from "../moment/validateDate";
+
+export const ValidateFields = (data) =>{
+
+    const errors = {};
+    const emailRegex = new RegExp('[a-z0-9]+@');
+    const ssnRegex = new RegExp('[0-9]{3}-[0-9]{2}-[0-9]{4}'); 
+
+    if(data.firstName.trim() === ''){
+        errors.firstName = true; 
+    }
+    if(data.lastName.trim() === ''){
+        errors.lastName = true; 
+    }
+    if(data.emailAddress.trim() === '' || !emailRegex.test(data.emailAddress)){
+        errors.emailAddress = true; 
+    }
+    if(data.address.trim() === ''){
+        errors.address = true; 
+    }
+    if(data.password.length < 6){
+        errors.password = true; 
+    }
+    if(!validateDate(data.dob)){
+        errors.dob = true; 
+    }
+    if(data.workPhone.trim() === ''){
+        errors.workPhone = true; 
+    }
+    if(data.locality.trim() === ''){
+        errors.locality = true; 
+    }
+    if(data.mobilePhone.trim() === ''){
+        errors.mobilePhone = true; 
+    }
+    if(data.region.trim() === ''){
+        errors.region = true; 
+    }
+    if(data.ssn.trim() === '' || !ssnRegex.test(data.ssn)){
+        errors.ssn = true; 
+    }
+    if(data.homePhone.trim() === ''){
+        errors.homePhone = true; 
+    }
+    if(data.postalCode.trim() === ''){
+        errors.postalCode = true; 
+    }
+    if(data.country.trim() === ''){
+        errors.country = true; 
+    }
```

Comment:
If you add the `required` attribute to the corresponding HTML elements, you can get rid of repeated validations and just leave Email and SSN validations.

Reply:
Also, you don't have to worry about empty field messages. The `required` attribute does that for you :)

<img width="246" alt="Screen Shot 2022-10-19 at 4 17 59 PM" src="https://user-images.githubusercontent.com/96072727/196788257-210d4c99-e059-40c1-972c-d7f72053043f.png">

Reply:
Would that prevent the user from enter empty spaces?

Reply:
We talked about it through Slack!